### PR TITLE
W-14881103 xml sdk tls j17

### PR DESCRIPTION
**Pull request by sathya0**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/xml-sdk.adoc [hunk 1]
```diff
@@ -1211,6 +1211,78 @@ To use the operations from the <<example2, example>> above in a Mule app, it is
...
 ----
 
+[[xml-sdk-tls]]
+== TLS Support
+
+Starting with Mule 4.6, XML SDK supports Transport Layer Security (TLS). Use TLS to enhance security for your app. XML SDK connectors can support the `<tls:context>` parameter in their configuration parameters.
+
+=== Parameter Declaration
+
+Use the `xmlns:tlsEnabled=true` annotation to indicate the component to support TLS. The annotated element does not necessarily have to be a top level element. In the following example, the annotated element is `<http:request-connection>` which is a child of the `<http:request-config>` top level element.
```
Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]:
```suggestion
Use the `xmlns:tlsEnabled=true` annotation to indicate the component to support TLS. The annotated element does not necessarily have to be a top-level element. In the following example, the annotated element is `<http:request-connection>` which is a child of the `<http:request-config>` top-level element:
```

Comment (hannanelson) [reply modules/ROOT/pages/xml-sdk.adoc]: Does the annotation turn on TLS support or is it just a flag that says it updates TLS support?

File: modules/ROOT/pages/xml-sdk.adoc [hunk 2]
```diff
@@ -1211,6 +1211,78 @@ To use the operations from the <<example2, example>> above in a Mule app, it is
...
+[source,xml,linenums]
+----
+<http:request-config
+name="concur-httpreq-config" xmlns:connection="true"
+basePath ....>
+  <http:request-connection xmlns:tlsEnabled="true"
+  host=...>
+----
+
+The `<tls:context>` parameter can be specified only in one component. Using it on more than one component results in an error.
```
Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]: Any way to make active?

Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]:
```suggestion
The `<tls:context>` parameter can be specified in only one component. Using it on more than one component results in an error. 
```

File: modules/ROOT/pages/xml-sdk.adoc [hunk 3]
```diff
@@ -1211,6 +1211,78 @@ To use the operations from the <<example2, example>> above in a Mule app, it is
...
+basePath ....>
+  <http:request-connection xmlns:tlsEnabled="true"
+  host=...>
+----
+
+The `<tls:context>` parameter can be specified only in one component. Using it on more than one component results in an error. 
+
+XML SDK adds a `<tls:context>` parameter in the configuration of the generated extension. This parameter is optional unless the `<tls:context>` parameter in the target element is required and doesn’t have a default value.
+
+When the extension is used, the supplied `<tls:context>` parameter is mapped to the config element with this annotation.
```
Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]: Active voice? Who/what uses the extension? (Will the audience know?)

(It's hard to know who's doing what here.)

File: modules/ROOT/pages/xml-sdk.adoc [hunk 4]
```diff
@@ -1211,6 +1211,78 @@ To use the operations from the <<example2, example>> above in a Mule app, it is
...
+  host=...>
+----
+
+The `<tls:context>` parameter can be specified only in one component. Using it on more than one component results in an error. 
+
+XML SDK adds a `<tls:context>` parameter in the configuration of the generated extension. This parameter is optional unless the `<tls:context>` parameter in the target element is required and doesn’t have a default value.
+
+When the extension is used, the supplied `<tls:context>` parameter is mapped to the config element with this annotation. 
+
+==== Inline Declaration
```
Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]: Consider making these H4s a bullet list or DL.

File: modules/ROOT/pages/xml-sdk.adoc [hunk 5]
```diff
@@ -1211,6 +1211,78 @@ To use the operations from the <<example2, example>> above in a Mule app, it is
...
+basePath ....>
+  <http:request-connection xmlns:tlsEnabled="true"
+  host=...>
+	  <tls:context>
+      <tls:trust-store path="ssltest-cacerts.jks" password="changeit2"/>
+    </tls:context>a
+  </http:request-connection>
+----
+
+In this case, the value that comes from the module definition is used as the default. The synthetic parameter is still added to the module’s config, which, if provided, is used as an override. Providing a default value at the target element makes the synthetic parameter optional even if the parameter at the target element is required. This applies to both inline and by-reference declarations.
```
Comment (hannanelson) [inline modules/ROOT/pages/xml-sdk.adoc]: still -> also?

Comment (hannanelson) [reply modules/ROOT/pages/xml-sdk.adoc]: "This" 
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22it,%22%22this,%22%22these%22

File: modules/ROOT/pages/xml-sdk.adoc (hunk 1 above)
Comment (sathya0) [reply modules/ROOT/pages/xml-sdk.adoc]: It signals the component to TLS should be delegated to. So the first one?

Comment (hannanelson) [reply modules/ROOT/pages/xml-sdk.adoc]: Should it say "to specify that the component support TLS"?

**Review by hannanelson — DISMISSED** [review]

File: modules/ROOT/pages/xml-sdk.adoc (hunk 1 above)
Comment (hannanelson) [reply modules/ROOT/pages/xml-sdk.adoc]: Or
Assign the xxx annotation to the component to support TLS. 
Or
Assign the xxx annotation at the component level to enable support for TLS.

**Review by hannanelson — APPROVED** [review]

**Pull request merged**